packages/trpc/server/routers/viewer/slots/util.ts

```diff
@@ -344,13 +345,69 @@ export async function getAvailableSlots({ input, ctx }: GetScheduleOptions) {
   }
   let currentSeats: CurrentSeats | undefined;
 
-  let usersWithCredentials = eventType.users.map((user) => ({
-    isFixed: !eventType.schedulingType || eventType.schedulingType === SchedulingType.COLLECTIVE,
-    ...user,
-  }));
-  // overwrite if it is a team event & hosts is set, otherwise keep using users.
-  if (eventType.schedulingType && !!eventType.hosts?.length) {
-    usersWithCredentials = eventType.hosts.map(({ isFixed, user }) => ({ isFixed, ...user }));
+  let usersWithCredentials;
+  let teamMember;
+
+  if (eventType.schedulingType === SchedulingType.ROUND_ROBIN && input.bookerEmail) {
+    let crmRoundRobinLeadSkip;
+    // See if CRM app is enabled and skip RR assignment
+    const eventTypeAppMetadata = eventType?.metadata?.apps;
+    for (const appKey in eventTypeAppMetadata) {
+      const app = eventTypeAppMetadata[appKey];
+      if (
+        app.enabled &&
+        app.appCategories &&
+        app.appCategories.some((category) => category === "crm") &&
+        app.roundRobinLeadSkip
+      ) {
+        crmRoundRobinLeadSkip = app;
+        break;
+      }
+    }
+
+    if (crmRoundRobinLeadSkip) {
+      const crmCredential = await prisma.credential.findUnique({
+        where: {
+          id: crmRoundRobinLeadSkip.credentialId,
+        },
+        include: {
+          user: {
+            select: {
+              email: true,
+            },
+          },
+        },
+      });
+      if (crmCredential) {
+        const crm = new CrmManager(crmCredential);
+        const contact = await crm.getContacts(input.bookerEmail, true);
```

Comment:
I was scared about making one more call when loading slots but if we narrow down the host we only need to get slots for that one host vs all hosts under an event type.